Source/WebKit/Platform/IPC/Encoder.cpp

```diff
@@ -141,24 +144,21 @@ static inline size_t roundUpToAlignment(size_t value, size_t alignment)
 
 void Encoder::reserve(size_t size)
 {
-    if (size <= m_bufferCapacity)
+    if (size <= m_capacityBuffer.size())
         return;
 
-    size_t newCapacity = roundUpToAlignment(m_bufferCapacity * 2, 4096);
+    size_t newCapacity = roundUpToAlignment(m_capacityBuffer.size() * 2, 4096);
     while (newCapacity < size)
         newCapacity *= 2;
 
-    uint8_t* newBuffer;
-    if (!allocBuffer(newBuffer, newCapacity))
+    auto newBuffer = allocBuffer(newCapacity);
+    if (newBuffer.empty())
```

Comment:
Not sure why `allocBuffer` itself doesn't do the `CRASH` on failure. Do you see a reason?

Reply:
Yes, I think `allocBuffer` can CRASH on failure. I'll update.